src/wasm/wasm-s-parser.cpp

```diff
@@ -1062,13 +1062,24 @@ Expression* SExpressionWasmBuilder::makeBlock(Element& s) {
     }
     curr->name = nameMapper.pushLabelName(sName);
     if (i >= s.size()) break; // empty block
+
+    // block signature
+    // TODO(sbc): Remove support for old result syntax once the spec tests
+    // are updated.
     if (s[i]->isStr()) {
-      // block signature
       curr->type = stringToWasmType(s[i++]->str());
-      if (i >= s.size()) break; // empty block
     } else {
-      curr->type = none;
+      Element& params = *s[i];
+      IString id = params[0]->str();
+      if (id == RESULT) {
+        curr->type = stringToWasmType(params[1]->str());
```

Comment:
Might be worth refactoring that, but `get` seems too strong a name, since it's checking if something is a result. If it's not a result, it returns none correctly, but it didn't "get" that from the input, if you see what I mean.

Maybe I'm overthinking it, lgtm (for the PR as a whole) either way.